### PR TITLE
[Podspec] Fix podspec warnings when issuing a lint

### LINKED ISSUE #2804: Fix podspec warnings
pod spec lint MaterialComponents.podspec returns warnings.

### PR DESCRIPTION
**Pull request by yarneo**
Mostly suppressing partial-availability warnings with clang. One image didn't have the right color depth.

This will allow us to issue a `pod lib lint` without `--allow-warnings` and pass validation.
  Closes #2804

### CODE REVIEW

File: components/Typography/src/MDCTypography.m [hunk 1]
```diff
@@ -344,6 +356,7 @@ - (BOOL)isLargeForContrastRatios:(UIFont *)font {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wtautological-pointer-compare"
 #pragma clang diagnostic ignored "-Wunreachable-code"
+#pragma clang diagnostic ignored "-Wpartial-availability"
```
Comment (ianegordon) [inline components/Typography/src/MDCTypography.m:359]: Should we be clang diagnostic pop after the call below?

**Review by ianegordon — APPROVED** [review]
Please address comment before merge, but LGTM!

File: components/Typography/src/MDCTypography.m (hunk 1 above)
Comment (romoore) [reply components/Typography/src/MDCTypography.m:359]: Looks like there was one already there.

Comment (yarneo) [reply components/Typography/src/MDCTypography.m:359]: yep!

File: components/Typography/src/private/UIFont+MaterialTypographyPrivate.m [hunk 2]
```diff
@@ -21,6 +21,8 @@ @implementation UIFont (MaterialTypographyPrivate)
 + (NSString *)mdc_fontWeightDescription:(CGFloat)weight {
 // The UIFontWeight enumeration was added in iOS 8.2
 #if defined(__IPHONE_8_2)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpartial-availability"
   if (weight == UIFontWeightUltraLight) {
```
Comment (romoore) [inline components/Typography/src/private/UIFont+MaterialTypographyPrivate.m]: Check if the symbol exists?

**Review by romoore — APPROVED** [review]
What changes with the reorder icon? They look the same.

**Comment by yarneo** [comment]
There was an issue with the previous icon that the PNG didn't have 8 bit input depth, and was causing a warning. it's great that they look the same :)

File: components/Typography/src/private/UIFont+MaterialTypographyPrivate.m (hunk 2 above)
Comment (yarneo) [reply components/Typography/src/private/UIFont+MaterialTypographyPrivate.m]: done

**Pull request merged**